library/spdm_responder_lib/libspdm_rsp_certificate.c

```diff
@@ -111,6 +113,36 @@ libspdm_return_t libspdm_get_response_certificate(libspdm_context_t *spdm_contex
         length = (uint16_t)(spdm_context->local_context.local_cert_chain_provision_size[slot_id] -
                             offset);
     }
+
+    /* In chunk case, ensure the length request can be served, otherwise return error. */
+    if (libspdm_is_capabilities_flag_supported(spdm_context, false,
```

Comment:
Why is this needed? Wouldn't `libspdm_build_response` handle chunking like for all the other messages?

Reply:
The length must be calculated and checked here properly.

Otherwise, it will cause ASSERT at line 152 - `LIBSPDM_ASSERT(*response_size >= sizeof(spdm_certificate_response_t) + length);`

Reply:
Ah so this is a case where the Integrator has provisioned a certificate chain that is larger than its `MaxSPDMmsgSize`. In that case this is true even if chunking is not enabled and the
```
LIBSPDM_ASSERT (length + sizeof(spdm_certificate_response_t) <=
                        spdm_context->local_context.capability.max_spdm_msg_size);
```
is unconditional, assuming that `max_spdm_msg_size != 0`.

Reply:
That check can also be added to `libspdm_check_context` to help them catch that sooner.

Reply:
For local_context, I dont believe we need 0 check.

Reply:
Uh, I am not sure if `libspdm_check_context` can help here, because the certificate chain may be provisioned later.
What we can do is only "if cert_chain != NULL, then check".

But we really dont know when `libspdm_check_context` will be invoked by the integrator. Maybe never.
Then we still need ASSERT here, even we add to `libspdm_check_context`

Reply:
The assertion would remain here but it can also be checked in `libspdm_check_context`.

Reply:
agree. Done.